components/uniset/src/bin/gen_properties.rs

```diff
@@ -0,0 +1,458 @@
+use std::char;
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fs::File;
+use std::io::{self, BufRead, Error};
+use std::iter::Iterator;
+use std::panic;
+use std::path::Path;
+use std::u32;
+
+use icu_unicodeset::{UnicodeSet, UnicodeSetBuilder};
+
+//
+// Can run with command in root of icu_unicodeset crate:
+//   cargo run --bin genprops <ppucd-file>
+//
+
+fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
+where
+    P: AsRef<Path>,
+{
+    let file = File::open(filename)?;
+    Ok(io::BufReader::new(file).lines())
+}
+
+fn get_line_str_opt(line_result: Result<String, Error>) -> Option<String> {
```

Comment:
This is equivalent to `line_result.ok()`

https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

Reply:
Done.